Core/include/Acts/EventData/detail/TestSourceLink.hpp

```diff
@@ -87,17 +92,40 @@ struct TestSourceLink final {
     return os;
   }
   constexpr std::size_t index() const { return sourceId; }
+};
+
+struct TestSourceLinkSurfaceAccessor {
+  const TrackingGeometry* geometry = nullptr;
 
-  struct SurfaceAccessor {
-    const Acts::TrackingGeometry& trackingGeometry;
```

Comment:
Sorry one more thing - I would also revert to the reference to make the change less intrusive